packages/vue-app/template/components/nuxt-link.js

```diff
@@ -1,8 +1,34 @@
 <%= isTest ? '// @vue/component' : '' %>
+import Vue from 'vue'
+
 export default {
+  extends: Vue.component('RouterLink'),
   name: 'NuxtLink',
-  functional: true,
-  render(h, { data, children }) {
-    return h('router-link', data, children)
+  props: {
+    noPrefetch: {
+      type: Boolean,
+      default: false
+    }
+  },
+  mounted() {
+    if (!this.noPrefetch) this.$nextTick(this.prefetch)
```

Comment:
I'd suggest to better stick to a curly if here, especially bc we don't have a fully working autocompletion in the template file.
```suggestion
    if (!this.noPrefetch){
      this.$nextTick(this.prefetch)
    }
```